hathor/verification/transaction_verifier.py

```diff
@@ -246,19 +248,38 @@ def verify_sum(self, token_dict: dict[TokenUid, TokenInfo]) -> None:
                 pass
             elif token_info.amount < 0:
                 # tokens have been melted
-                if not token_info.can_melt:
+                if not token_info.can_melt and token_info.version != TokenVersion.DEPOSIT:
```

Comment:
This logic seems right but it is confusing. I'd rather have a match for token version with each verification for each type separately.